src/rules.ts

```diff
@@ -186,7 +186,9 @@ function evaluateCondition(subjectAttributes: Record<string, any>, condition: Co
         }
         return compareNumber(value, condition.value, (a, b) => a < b);
       case OperatorType.MATCHES:
-        return new RegExp(condition.value as string).test(value as string);
+        return new RegExp(`^${condition.value as string}$`).test(value as string);
```

Comment:
Note this will break many user's existing targeting rules (For example taxfix's check for `taxfix` in email).